spec/unit/util/http_proxy_spec.rb

```diff
@@ -79,8 +79,7 @@ def expects_proxy_connection_via(http, www, host, port, user, password)
         expects_direct_connection_to(http, www)
       end
 
-      it 'connects directly to the server when no_proxy matches domain like ruby does' do
-        pending("PUP-10106 Puppet doesn't match domains like ruby")
+      it 'connects directly to the server when no_proxy matches a domain suffix like ruby does' do
```

Comment:
Do you think it would be a good idea to add a test to verify the additional matching we now expect? ie example.com would be expected to match with `Puppet[:no_proxy] = 'ample.com'`?